content/articles/spotting-of-passing-objects-using-python/index.md

```diff
@@ -0,0 +1,165 @@
+ As you have been following, let's repeat the procedure in the above commands.
+ ```bash
+  pip install imutils
+ ```
+After successful installations, we will now jump into the code area.
+### Case 1:Importing the external libraries.
+As we had discussed earlier the functions of these libraries, we will be now importing them for use in our program.
+```python
+import cv2  
+import numpy as np  
+import datetime  
+import imutils
+```
+Datetime module is called, as it will be used for returning the live or real-time time and date in our footage.
+
+### Case 2:Variables initialization
+This is achieved by setting a variable equal to a given value, thus the variable is created. So in our case, we will be able to use the variables just by calling them.
+```python
+rec=cv2.VideoCapture(0)  
+sto, mapping1=rec.read()  
+sto, mapping2=rec.read()
+```
+In the function VideoCapture, we use `0` since it is the default value for the webcam.
+### Case 3:Detection and conversion of the frame to grayscale
+Video is a series of continuous images with the difference in time. Likewise, it's a slide show of images. So we will have to initially identify and recognize the frames then have them processed. The frame is converted to grayscale to increase the accuracy of key features detection, eliminating possible misses.
+Since python language is feature sensitive, take care of your indentations.  
+```python
+while rec.isOpened():  
+    sub=cv2.absdiff(mapping1, mapping2)  
+    convclr=cv2.cvtColor(sub, cv2.COLOR_BGR2GRAY)  
+    blur=cv2.GaussianBlur(convclr, (3, 3), 1)  
+    _, thresh=cv2.threshold(blur, 20, 255, cv2.THRESH_BINARY)  
+    enlarged=cv2.dilate(thresh, None, iterations=1)  
+    contours, _=cv2.findContours(enlarged, cv2.RETR_TREE, cv2.CHAIN_APPROX_SIMPLE)
+```
+### Case 4:Blurring the frame
+Calling from cv2, we have a module responsible for performing blurring. The Gaussian blur. It smoothens and average pixel intensity across pixel height against its width, thus filtering high-intensity noise. Let's see how to implement it below.
+```python
+	if mapping1 is None:  
+	    break  
+    
+	mapping1 = imutils.resize(mapping1, width=700)  
+	convclr1 = cv2.cvtColor(mapping1, cv2.COLOR_BGR2GRAY)  
+	convclr2 = cv2.GaussianBlur(convclr1,(21, 21), 0)
+```
+Using `imutils` we can resize our frame to the desired size.
+
+### Case 5:Finding the difference of the delta frames
+Since we stored the first frame in mapping1, we will be able to compute the difference using the later frames stored in mapping2.
+```bash
+	if mapping1 is None:  
+	    mapping1=convclr2  
+	    continue  
+
+	frameDelta = cv2.absdiff(convclr2, convclr1)  
+	thresh = cv2.threshold(frameDelta, 25, 255, cv2.THRESH_BINARY)[1]
+```
+### Case 6: Enlarging the images in the footage.
+To enlarge the images, we call a dilate function from cv2. This function fills in the holes and finds contours on the margin image.
```

Comment:
Why are we enlarging it?

`This function fills in the holes and finds contours on the margin image.`

How?